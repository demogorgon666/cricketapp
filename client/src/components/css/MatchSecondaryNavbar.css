@import url("https://fonts.googleapis.com/css?family=Montserrat:400,500,600&display=swap");

.secNavParent {
	width: 100vw;
	background-color: #ffffff;
	box-shadow: 0px 2px 4px 0px #e6e6e6;
	color: #2c3e50;
	display: flex;
	flex-direction: column;
	letter-spacing: 0.3px;
	position: fixed;
	font-family: "Montserrat", sans-serif;
	top: 60px;
	left: 0;
	right: 0;
  }
  
  .secNavHead {
	display: flex;
	flex-direction: row;
	align-items: center;
	margin: 14px 60px 10px 60px;
  }
  
  .secNavHead div:first-child {
	display: inline-block;
	width: 60px;
	height: 60px;
	border-radius: 50%;
	background-repeat: no-repeat;
	background-position: center center;
	background-size: cover;
  }
  
  .secNavHead div:last-child {
	display: flex;
	flex-direction: column;
	margin-left: 40px;
  }
  
  .secNavHead div:last-child h2,
  .secNavHead div:last-child h4 {
	margin: 0px;
  }
  
  .secNavHead div:last-child h4 {
	color: #707070;
	font-weight: 400;
  }
  
  .secNavLinks {
	margin: 0px 60px;
  }
  
  .secNavLinks ul {
	display: flex;
	align-items: center;
	list-style-type: none;
	margin: 0px;
	padding: 0px;
  }
  
  .secNavLinks ul li {
	font-family: "Montserrat", sans-serif;
	padding: 10px 0px;
	transition: 0.3s;
	color: #272727;
	margin-right: 30px;
	letter-spacing: 0.3px;
	text-align: center;
  }
  
  .secNavLinks ul li:hover {
	font-weight: 600;
	cursor: pointer;
	padding-bottom: 7px;
	color: #2980b9;
	border-bottom: 3px solid #2980b9;
  }
  
  .secLink {
	text-decoration: none;
  }
  
.summary{
    margin: 20px;
    margin-top: 80px;
}
.top-div-of-match {
    display: flex;
    flex-direction: row;
    width: 95%;
    height: auto;
}

.short-summary-with-result{
    width: 40%;
    border-right: 1px solid #272727;
    margin-left: 20px;
}
.short-summary-with-result span{
text-align: left;
margin-bottom: 20px;    
}

.short-summary-with-player-of-the-match{
width: 50%;
border-left: 1px solid #272727;
}
.short-summary-right .default-img{
    height: 60px;
    width: 120px;
    background-size: 120px 65px;
    margin-top: 15px;

}

.short-summary-with-result .Team-data {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    text-align: left;
    font-size: 20px;
    margin: 3px;
    margin-left: 10px;
}

.Team-data .Team-name , .Team-score{
    text-align: left;
    font-size: 20px;
    font-weight: 500;
    margin-right: 40px;
    margin-top: 10px;
    }
.Team-data .Team-name , .Team-score{
    text-align: left;
    font-size: 20px;
    font-weight: 500;
    margin-right: 40px;
    margin-top: 10px;
    }
.Team-data .Team-img {
    width: 60px;
    background-size: 60px 34px;
}

.short-summary-right{
    display: grid;
    grid-template-columns: 20% 50%;
    margin-left: 20px;
    height: 50px;

}
.header-player-of-the-match{
    margin: 40px 0 10px 40px;
}



@media only screen and (max-width: 576px) {
	.secNavParent {
		padding: 20px 20px 0px 20px;
	}
	.secNavLinks ul li {
		font-size: 12px;
	}
	.secNavHead div:first-child {
		width: 60px;
		height: 60px;
	}
	.secNavLinks {
		margin-top: 10px;
	}
	.secNavHead div:last-child {
		margin-left: 20px;
	}
}
