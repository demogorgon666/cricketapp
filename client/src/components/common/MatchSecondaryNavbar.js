import React, { Component } from "react";
import "../css/MatchSecondaryNavbar.css";
import "../css/SecondaryNavbar.css";
import south_africa from "../images/SouthAfrica.jpeg";
import india from "../images/india.jpeg";
export default class MatchSecondaryNavbar extends Component {
  state = {
    summary: true,
    scorecard: false,
    stats: false
  }
  render() {
    return (
      <div>
        <div className="secNavParent">
          <div className="matchSecNavHead">
            <div className="match-score-details">
              <span id="match-result" style={{ fontSize: "14px" }}>
                Result:{" "}
                <b id="match-winner-name-runs">
                  {this.props.match.team_winner} {this.props.match.won_by}
                </b>
              </span>
              <div className="match-team-score">
                <div>
                  <div
                    style={{
                      backgroundImage: `url(data:image/jpeg;base64,${this.props.match.team_one_img}`
                    }}
                  ></div>
                  <div className="team-name">{this.props.match.teamOne}</div>
                  <div className="team-score">
                    {" "}
                    {this.props.match.teamOneScore}/
                    {this.props.match.teamone_wicket} (
                    {this.props.match.team_one_total_over}.{this.props.match.team_one_total_ball} overs)
                  </div>
                </div>
                <div>
                  <div
                    style={{
                      backgroundImage: `url(data:image/jpeg;base64,${this.props.match.team_two_img}`
                    }}
                  ></div>
                  <div className="team-name">{this.props.match.teamTwo}</div>
                  <div className="team-score">
                    {" "}
                    {this.props.match.teamTwoScore}/
                    {this.props.match.teamtwo_wicket} (
                    {this.props.match.team_two_total_over}.{this.props.match.team_two_total_ball} overs)
                  </div>
                </div>
              </div>
            </div>
            <div className="match-pom">
              <span style={{ fontSize: "14px" }}>Player of the match</span>
              <div className="pom-name-photo">
                <div className="pom-img">
                  <i className="fas fa-user"></i>
                </div>
                <div className="pom-name">
                  <div>{this.props.match.player_of_the_match}</div>
                  <div>{this.props.match.team_winner}</div>
                </div>
              </div>
            </div>
          </div>
          <div className="secNavLink">
            <ul>
              <span
                id="summary-click"
                className="secLink"
                onClick={() => {
                  this.setState({ summary: true, scorecard: false, stats: false })
                  this.props.changeDetailsType("summary");
                }}
              >
                <li id="match-summary" className={this.state.summary ? "options-selected-li" : "options-li"}>Summary</li>
              </span>
              <span
                className="secLink"
                onClick={() => {
                  this.setState({ summary: false, scorecard: true, stats: false })
                  this.props.changeDetailsType("scorecard");
                }}
              >
                <li id="match-scorecard" className={this.state.scorecard ? "options-selected-li" : "options-li"}>ScoreBoard</li>
              </span>
              <span
                className="secLink"
                onClick={() => {
                  this.setState({ summary: false, scorecard: false, stats: true })
                  this.props.changeDetailsType("stats");
                }}
              >
                <li id="match-stats" className={this.state.stats ? "options-selected-li" : "options-li"}>Statistics</li>
              </span>
            </ul>
          </div>
        </div>
      </div>
    );
  }
}
