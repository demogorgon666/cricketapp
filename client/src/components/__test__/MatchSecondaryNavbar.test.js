import React from "react";
import { shallow } from "enzyme";
import MatchSecondaryNavbar from "../common/MatchSecondaryNavbar";

const wrapper = shallow(<MatchSecondaryNavbar />);

describe("[MatchSecondaryNavbar] test for the text,css properties", () => {
	it("should have summary, scoreboard, stats", () => {
		// checks for no. of link present on navbar
		expect(wrapper.find(".link").length).toBe(3);

		// checks for the links text
		expect(
			wrapper
				.find(".link")
				.at(0)
				.text()
		).toBe("Summary");
		expect(
			wrapper
				.find(".link")
				.at(1)
				.text()
		).toBe("ScoreBoard");
		expect(
			wrapper
				.find(".link")
				.at(2)
				.text()
		).toBe("Statistics");
	});

	it("should have one h2 header", () => {
		expect(wrapper.find("h2").length).toBe(1);
	});
	it("should have one div tag with 'Player of the match", () => {
		expect(wrapper.find(".short-summary-with-player-of-the-match span").text()).toBe("Player of the match");
	});
	it("should have one h2 content with 'Recent Matches' content", () => {
		expect(wrapper.find(".h2-recent-matches").text()).toBe("MS Dhoni");
	});
});
