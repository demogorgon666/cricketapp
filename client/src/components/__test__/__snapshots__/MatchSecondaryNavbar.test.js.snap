// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[MatchSecondaryNavbar] test renders the component for snapshot testing 1`] = `
<div>
  <div
    className="secNavParent"
  >
    <div
      className="matchSecNavHead"
    >
      <div
        className="match-score-details"
      >
        <span
          id="match-result"
          style={
            Object {
              "fontSize": "14px",
            }
          }
        >
          Result:
           
          <b
            id="match-winner-name-runs"
          >
             
          </b>
        </span>
        <div
          className="match-team-score"
        >
          <div>
            <div
              style={
                Object {
                  "backgroundImage": "url(india.jpeg)",
                }
              }
            />
            <div
              className="team-name"
            />
            <div
              className="team-score"
            >
               
              /
               (
               overs)
            </div>
          </div>
          <div>
            <div
              style={
                Object {
                  "backgroundImage": "url(SouthAfrica.jpeg)",
                }
              }
            />
            <div
              className="team-name"
            />
            <div
              className="team-score"
            >
               
              /
               (
               overs)
            </div>
          </div>
        </div>
      </div>
      <div
        className="match-pom"
      >
        <span
          style={
            Object {
              "fontSize": "14px",
            }
          }
        >
          Player of the match
        </span>
        <div
          className="pom-name-photo"
        >
          <div
            className="pom-img"
          >
            <i
              className="fas fa-user"
            />
          </div>
          <div
            className="pom-name"
          >
            <div />
            <div />
          </div>
        </div>
      </div>
    </div>
    <div
      className="secNavLinks"
    >
      <ul>
        <span
          className="secLink"
          id="summary-click"
          onClick={[Function]}
        >
          <li
            id="match-summary"
          >
            Summary
          </li>
        </span>
        <span
          className="secLink"
          onClick={[Function]}
        >
          <li
            id="match-scorecard"
          >
            ScoreBoard
          </li>
        </span>
        <span
          className="secLink"
          onClick={[Function]}
        >
          <li
            id="match-stats"
          >
            Statistics
          </li>
        </span>
      </ul>
    </div>
  </div>
</div>
`;
